diff --git a/modules/cgal/src/internal/polyhedrons.cpp b/modules/cgal/src/internal/polyhedrons.cpp
index 7f1a355f1c..e8327312b2 100644
--- a/modules/cgal/src/internal/polyhedrons.cpp
+++ b/modules/cgal/src/internal/polyhedrons.cpp
@@ -171,7 +171,11 @@ std::pair<algebra::Vector3Ds, Ints> get_indexed_facets(
 }
 
 template <class K>
+#if CGAL_VERSION_NR >= 1050421000
+CGAL::Polyhedron_3<K> create_cube(const algebra::BoundingBoxD<3> &bb) {
+#else
 CGAL::Nef_polyhedron_3<K> create_cube(const algebra::BoundingBoxD<3> &bb) {
+#endif
   typename CGAL::Polyhedron_3<K> p;
   // appends a cube of size [0,1]^3 to the polyhedron P.
   CGAL_precondition(p.is_valid());
